src/pages/Signup.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { FormEvent, useState } from "react";
+import { Link } from "react-router-dom";
+import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
+import TextInput from "../components/TextInput.component";
+import PasswordInput from "../components/PasswordInput.component";
+import { auth, db } from "../utils/firebase/firebase.utils";
+import { useNavigate } from "react-router-dom";
+import { doc, getDoc, setDoc } from "firebase/firestore";
+
+const Signup = () => {
+  const navigate = useNavigate();
+  const [values, setValues] = useState({
+    fullName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const [errorMsg, setErrorMsg] = useState("");
+  const [submitButtonDisabled, setSubmitButtonDisabled] = useState(false);
+
+  const handleChangeName = (e: FormEvent) => {
+    e.preventDefault();
+    setValues((prev) => ({
+      ...prev,
+      fullName: (e.target as HTMLInputElement).value,
+    }));
+  };
+  const handleChangeEmail = (e: FormEvent) => {
+    e.preventDefault();
+    setValues((prev) => ({
+      ...prev,
+      email: (e.target as HTMLInputElement).value,
+    }));
+  };
+  const handleChangePassword = (e: FormEvent) => {
+    e.preventDefault();
+    setValues((prev) => ({
+      ...prev,
+      password: (e.target as HTMLInputElement).value,
+    }));
+  };
+
+  const handleChangeConfirmPassword = (e: FormEvent) => {
+    e.preventDefault();
+    setValues((prev) => ({
+      ...prev,
+      confirmPassword: (e.target as HTMLInputElement).value,
+    }));
+  };
+
+  const handleSubmission = (e: FormEvent) => {
+    e.preventDefault();
+
+    // Validations
+
+    if (
```

Comment:
It's best to separate functionality into different functions. Although validating input is definitely a part of handling submission, it would be better if you create a separate function for handling validation.